apis/python/CMakeLists.txt

```diff
@@ -0,0 +1,332 @@
+cmake_minimum_required(VERSION 3.21)
+cmake_policy(SET CMP0148 NEW) 
+
+# Set macOS architecture before project() to avoid vcpkg warnings
+if(APPLE AND NOT DEFINED CMAKE_OSX_ARCHITECTURES)
+    # Detect system architecture
+    execute_process(
+        COMMAND uname -m
+        OUTPUT_VARIABLE DETECTED_ARCH
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    set(CMAKE_OSX_ARCHITECTURES "${DETECTED_ARCH}" CACHE STRING "macOS architecture")
+    message(STATUS "Auto-detected macOS architecture: ${CMAKE_OSX_ARCHITECTURES}")
+endif()
+
+project(tiledbsoma-python LANGUAGES CXX)
+
+# Set C++ standard
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Find pybind11 - required for pybind11_add_module
+find_package(pybind11 REQUIRED)
+find_package(Ninja)
+
+# Find Python - Use Development.Module (not Development) for manylinux compatibility
+# Development.Module only requires headers/metadata, not Python_LIBRARIES
+find_package(Python COMPONENTS Interpreter Development.Module REQUIRED)
+
+# Get version from scikit-build-core or setuptools_scm
+# scikit-build-core passes the version via SKBUILD_PROJECT_VERSION
+if(DEFINED SKBUILD_PROJECT_VERSION)
+    set(VERSION_STRING "${SKBUILD_PROJECT_VERSION}")
+    message(STATUS "Package version (from scikit-build-core): ${VERSION_STRING}")
+    
+    # Validate that setuptools_scm worked (version should not be unknown)
+    if(VERSION_STRING MATCHES "0\\.0\\.0\\+unknown")
+        message(FATAL_ERROR "setuptools_scm failed to determine version. Version is '${VERSION_STRING}'. "
+                "This likely means git tags are not available or the repository structure is incorrect.")
+    endif()
+elseif(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/RELEASE-VERSION")
+    # Fallback to RELEASE-VERSION file if it exists (for non-scikit-build builds)
+    file(READ "${CMAKE_CURRENT_SOURCE_DIR}/RELEASE-VERSION" VERSION_STRING)
+    string(STRIP "${VERSION_STRING}" VERSION_STRING)
+    message(STATUS "Package version (from RELEASE-VERSION): ${VERSION_STRING}")
+else()
+    # This should only happen in legacy builds or manual CMake invocations
+    set(VERSION_STRING "0.0.0.dev0")
+    message(WARNING "No version available. If building via pip/cibuildwheel, this indicates a problem.")
```

Comment:
Is there an environment variable we can use to fail if we are on pip/cibuildwheel?